hpat/tests/test_basic.py

```diff
@@ -204,18 +204,25 @@ def test_reduce(self):
             self.assertEqual(count_parfor_REPs(), 0)
 
     def test_array_reduce(self):
-        def test_impl(N):
-            A = np.ones(3);
-            B = np.ones(3);
-            for i in numba.prange(N):
-                A += B
-            return A
+        binops = ['+=', '*=', '+=', '*=', '|=', '|=']
+        dtypes = ['float32', 'float32', 'float64', 'float64', 'int32', 'int64']
+        for (op,typ) in zip(binops,dtypes):
+            func_text = """def f(n):
+                  A = np.arange(np.{}(10));
+                  B = np.arange(np.{}(3), np.{}(3+10));
```

Comment:
Please remove semicolons.
Also, I don't think the type of the generated arrays are correct. For example, `np.arange(np.float32(10))` will generate array of float64s instead of float32. The right way is to use `np.arange(0, 10, 1, np.float32)` (not using the dtype keyword argument since Numba doesn't support it).